tests/libcxx/host/host.cpp

```diff
@@ -1,22 +1,31 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
+#include <inttypes.h>
 #include <openenclave/host.h>
 #include <openenclave/internal/calls.h>
 #include <openenclave/internal/error.h>
 #include <openenclave/internal/tests.h>
-#include <pthread.h>
 #include <cassert>
+#include <chrono>
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <map>
 #include <thread>
+#include <vector>
 #include "libcxx_u.h"
 #include "threadArgs.h"
 
 // Host maintains a map of enclave key to host thread ID
-static std::map<uint64_t, pthread_t> _enclave_host_id_map;
+static std::map<uint64_t, size_t> _enclave_host_id_map;
+// Since C++ thread does not allow for using the built-in
+// thread id in thread operations such as thread referencing
+// and joining, we use customized host_thread_id to keep track
+// of created thread objects
+static std::vector<std::thread> _host_thread_list;
+static std::map<std::thread::id, size_t> _thread_host_id_map;
```

Comment:
This is all pretty convoluted. Why can't you just have `std::map<uint64_t, std::thread> _enclave_host_id_map;`? You don't really need the thread's id for any operations you're doing, just a reference to the object.

Reply:
The host thread routine (`host_enclave_thread`) need to retrieve the thread object of the thread itself. This can be done by both pthread (usiung `pthread_self())` and windows thread (get  `HANDLE` via thread id). However, it seems c++ thread does not allow that.

Using `map` may not be sufficient as we cannot obtain the `thread::id` before the thread creation.

Reply:
@jhand2 check again?

Reply:
Thanks for cleaning this up.

Can you squash the PR into one commit?

Reply:
done